src/Functions/FunctionsConversion.h

```diff
@@ -1860,6 +1864,10 @@ class FunctionConvertFromString : public IFunction
 
             if constexpr (std::is_same_v<ToDataType, DataTypeDateTime>)
                 res = std::make_shared<DataTypeDateTime>(extractTimeZoneNameFromFunctionArguments(arguments, 1, 0));
+            else if constexpr (std::is_same_v<ToDataType, DataTypeDate>)
+                res = std::make_shared<DataTypeDate>();
```

Comment:
Why this is needed? There is a final case, which covers it:
https://github.com/ClickHouse/ClickHouse/blob/908505c12e4a18c9945e51861dea7d3efb537068/src/Functions/FunctionsConversion.h#L1872-L1873